source/write/replace.txt

```diff
@@ -0,0 +1,193 @@
+.. _pymongo-write-replace:
+
+=================
+Replace Documents
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: modify, change, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use {+driver-short+} to perform a replace
+operation on a document in a MongoDB collection. A replace operation performs
+differently than an update operation. An update operation
+modifies only the specified fields in a target document. A replace operation removes *all* fields
+in the target document and replaces them with new ones.
+
+To learn more about update operations, see the :ref:`Update Documents guide.
+<pymongo-write-update>`
+
+.. .. tip:: Interactive Lab
+   
+..    This page includes a short interactive lab that demonstrates how to
+..    replace data by using the ``replace_one()`` method. You can complete this
+..    lab directly in your browser window without installing MongoDB or a code editor.
+
+..    To start the lab, click the :guilabel:`Open Interactive Tutorial` button at the
+..    top of the page. To expand the lab to a full-screen format, click the
+..    full-screen button (:guilabel:`⛶`) in the top-right corner of the lab pane.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<pymongo-get-started>` tutorial.
+
+Replace Operation
+-----------------
+
+You can perform a replace operation in MongoDB with the ``replace_one()`` method. 
+This method removes all fields except the ``_id`` field from the first document that
+matches the search criteria. It then inserts the fields and values you specify into the 
+document.
+
+Required Parameters
+~~~~~~~~~~~~~~~~~~~
+
+The ``replace_one()`` method requires the following parameters:
+
+- A **query filter** document, which determines which documents to replace. For
+  more information about query filters, see the 
+  :manual:`Query Filter Documents section </core/document/#query-filter-documents>` in
+  the MongoDB Server manual. 
+- A **replace** document, which specifies the fields and values to insert in the new 
+  document.
+
+Replace One
+-----------
+
+The following example uses the ``replace_one()`` method to replace the fields and values of a
+document with a ``name`` field value of "Pizza Town":
```

Comment:
S: I think it's clearer when these string values are also monospaced (`"Pizza Town"`).  Applies throughout, if you agree

Reply:
Yeah, I debated on doing it that way, especially since it's weird putting punctuation outside of the quotes. I'll change them to monospace

Reply:
Yeah, I debated on doing it that way, especially since it's weird putting punctuation outside of the quotes. I'll change them to monospace